include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -973,6 +978,261 @@ namespace internal
+      std::vector<double> &                       values,
+      const VectorType &                          vector,
+      const std::vector<types::global_dof_index> &indices,
+      const ComponentExtractor                    extract_component)
+    {
+      for (unsigned int i = 0; i < values.size(); ++i)
+        values[i] = get_component(vector[indices[i]], extract_component);
+    }
+
+
+    //    template <>
+    //    void
+    //    temp_get_indices<Vector<std::complex<double>>>::extract(
+    //      std::vector<double> &               values,
+    //      const Vector<std::complex<double>> &vector,
+    //      const std::vector<unsigned int> &   indices)
+    //    {
+    //      Assert(false, ExcNotImplemented());
+    //      // TODO
+    //    }
+    //    template <>
+    //    void
+    //    temp_get_indices<Vector<std::complex<float>>>::extract(
+    //      std::vector<double> &               values,
+    //      const Vector<std::complex<float>> &vector,
+    //      const std::vector<unsigned int> &   indices)
+    //    {
+    //      Assert(false, ExcNotImplemented());
+    //      // TODO
+    //    }
+
+    template <>
+    inline void
+    temp_get_indices<LinearAlgebra::EpetraWrappers::Vector>::extract(
+      std::vector<double> &                        values,
+      const LinearAlgebra::EpetraWrappers::Vector &vector,
+      const std::vector<types::global_dof_index> & indices,
+      const ComponentExtractor /*extract_component*/)
+    {
+      // I don't have element access
+      Assert(false, ExcNotImplemented());
+      // TODO
+    }
+
+    template <typename DoFHandlerType, typename VectorType>
+    void
+    MGDataEntry<DoFHandlerType, VectorType>::get_function_values(
+      const FEValuesBase<DoFHandlerType::dimension,
+                         DoFHandlerType::space_dimension> &fe_patch_values,
+      const ComponentExtractor                             extract_component,
+      std::vector<double> &                                patch_values) const
+    {
+      (void)extract_component;
+      Assert(extract_component == ComponentExtractor::real_part,
+             ExcNotImplemented());
+
+      typename DoFHandlerType::level_cell_iterator dof_cell(
+        &fe_patch_values.get_cell()->get_triangulation(),
+        fe_patch_values.get_cell()->level(),
+        fe_patch_values.get_cell()->index(),
+        this->dof_handler);
+
+      const VectorType *vector = &((*vectors)[dof_cell->level()]);
+
+      const unsigned int dofs_per_cell =
+        this->dof_handler->get_fe()[0].dofs_per_cell;
+
+      std::vector<types::global_dof_index> dof_indices(dofs_per_cell);
+      dof_cell->get_mg_dof_indices(dof_indices);
+      std::vector<double> values(dofs_per_cell);
+      temp_get_indices<VectorType>::extract(values,
+                                            *vector,
+                                            dof_indices,
+                                            extract_component);
+
```

Comment:
This line is necessary (see also https://github.com/peterrum/dealii/tree/mg_data_out_clean):
```suggestion

      std::fill(patch_values.begin(), patch_values.end(), 0.0); // clear old values !
```